README-National-Cloud.md

```diff
@@ -0,0 +1,193 @@
+7. In the succeeding page, Find the *Application ID* value and record it for later. You'll need it to configure the Visual Studio configuration file for this project.
+8. Then click on *Settings*, and choose *Properties*.
+9. For the App ID URI, replace the guid in the generated URI 'https://<your_tenant_name>/<guid>', with the name of your service, for example, 'https://<your_tenant_name>/TodoListWebApp_MT' (replacing *<your_tenant_name>* with the name of your Azure AD tenant).
+10. From the Settings | Reply URLs page for your application, update the Reply URL for the application to be *https://localhost:44302/*
+11. From the Settings menu, choose Keys and add a key - select a key duration of either 1 year or 2 years. When you save this page, the key value will be displayed, copy, and save the value in a safe location - you will need this key later to configure the project in Visual Studio. This key value will not be displayed again, nor retrievable by any other means, so record it as soon as it is visible from the Azure Government portal.
+12. Configure Permissions for your application. To that extent, in the Settings menu, choose the 'Required permissions' section and then,
+   click on **Add**, then **Select an API**, and type `Microsoft Graph` in the textbox. Then, click on  **Select Permissions** and select **User.Read** and **User.Read.All**.
+
+### Step 3: Configure the sample to use your Azure AD tenant
+
+In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+Open the solution in Visual Studio to configure the projects
+
+#### Configure the service project
+
+1. Open the `TodoListWebApp\Web.Config` file
+2. Find the app key `ida:ClientId` and replace the existing value with the application ID (clientId) of the `TodoListWebApp_MT` application copied from the Azure portal.
+3. Find the app key `ida:ClientSecret` and replace the existing value with the key you saved during the creation of the `TodoListWebApp_MT` app, in the Azure portal.
+4. Find the app key `ida:RedirectUri` and replace the existing value with the base address of the TodoListWebApp_MT project (by default `https://localhost:44302/`).
+5. Find the app key `ida:AADInstance` and replace the existing value with the corresponding [Azure AD endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud#azure-ad-authentication-endpoints) for the national cloud you want to target.
+6. Find the app key `ida:GraphAPIEndpoint` and replace the existing value with the corresponding [Microsoft Graph endpoint](https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints) for the  national cloud you want to target.
+
+
+### Step 5: Run the sample
+
+Clean the solution, rebuild the solution, and run it. The sample implements two distinct tasks: the onboarding of a new customer and regular sign in & use of the application.
+
+#### Sign up
```

Comment:
Maybe because I'm from B2C but I don't like the use of the term "Sign up" in this sample.  "Get the app" or "Admin Consent" seem like better choices here.  The Azure AD Gallery says "Add an app for my organization to use".  Azure Marketplace says "Get it now"
https://azuremarketplace.microsoft.com/en-us/marketplace/apps

Do we want to talk about the checkbox that we added to the normal authorization code flow if you are an administrator?  Even if the application does not pass prompt=admin_consent, the v2 endpoint will detect if the user is an admin and show a consent dialog with a checkbox saying "consent for all users in my organization".  This should work with the normal "Sign In" button in this application.

Reply:
this is a v1.0 sample, Saeed.